### ISSUE TITLE
build errors on aarch64

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#506]
There are probably some #ifdefs in `floatobject.c` or thereabouts that need to be reinstated.

```
/build/cc4XZZWI.s: Assembler messages:
/build/cc4XZZWI.s:23: Error: unknown mnemonic `fnstcw' -- `fnstcw [sp,22]'
/build/cc4XZZWI.s:55: Error: unknown mnemonic `fldcw' -- `fldcw [sp,14]'
```

https://logs.nix.ci/?key=nixos/nixpkgs.70415&attempt_id=2fe084b4-0634-450b-87e1-bc29ac592a1e

**Comment by andychu** [comment]
It's related to

HAVE_GCC_ASM_FOR_X87

and

Py_FORCE_DOUBLE in `pymath.h`.  (Is that used in rounding ?)

I guess it has something to do with 80-bit floating point:

https://stackoverflow.com/questions/3206101/extended-80-bit-double-floating-point-in-x87-not-sse2-we-dont-miss-it

**Comment by andychu** [comment]
I think it's only used to check if something's infinity:

```
/* Py_IS_INFINITY(X)                                                                                                                                                                                                                          
 * Return 1 if float or double arg is an infinity, else 0.                                                                                                                                                                                    
 * Caution:                                                                                                                                                                                                                                   
 *    X is evaluated more than once.                                                                                                                                                                                                          
 *    This implementation may set the underflow flag if |X| is very small;                                                                                                                                                                    
 *    it really can't be implemented correctly (& easily) before C99.                                                                                                                                                                         
 *    Override in pyconfig.h if you have a better spelling on your platform.                                                                                                                                                                  
 *  Py_FORCE_DOUBLE is used to avoid getting false negatives from a                                                                                                                                                                           
 *    non-infinite value v sitting in an 80-bit x87 register such that                                                                                                                                                                        
 *    v becomes infinite when spilled from the register to 64-bit memory.                                                                                                                                                                     
 * Note: PC/pyconfig.h defines Py_IS_INFINITY as _isinf                                                                                                                                                                                       
 */                                                                                                                                                                                                                                           
#ifndef Py_IS_INFINITY                                                                                                                                                                                                                        
#  if defined HAVE_DECL_ISINF && HAVE_DECL_ISINF == 1                                                                                                                                                                                         
#    define Py_IS_INFINITY(X) isinf(X)                                                                                                                                                                                                        
#  else                                                                                                                                                                                                                                       
#    define Py_IS_INFINITY(X) ((X) &&                                   \                                                                                                                                                                     
                               (Py_FORCE_DOUBLE(X)*0.5 == Py_FORCE_DOUBLE(X)))                                                                                                                                                                
#  endif                                                                                                                                                                                                                                      
#endif    
```

**Comment by andychu** [comment]
Released with 0.7.pre6 : https://www.oilshell.org/release/0.7.pre6/
